rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py

```diff
@@ -0,0 +1,107 @@
+"""
+Test NoSuchBucket error thrown for non existing bucket while applying ACL
+
+Usage: test_acl_for_nonexisting_bucket.py -c configs/<input-yaml>
+where : <input-yaml> is test_acl_for_nonexisting_bucket.yaml
+"""
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import argparse
+import json
+import logging
+import time
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from botocore.exceptions import ClientError
+from v2.lib.exceptions import RGWBaseException
+from v2.lib.resource_op import Config
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import HttpResponseParser, RGWService
+
+log = logging.getLogger()
+TEST_DATA_PATH = None
+
+
+def test_exec(config):
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+
+    all_users_info = s3lib.create_users(1)
+    # authenticate
+    auth = Auth(all_users_info[0], ssl=config.ssl)
+    rgw_conn = auth.do_auth()
+
+    if config.test_ops.get("set_acl", None) is True:
+        try:
+            s3_obj_acl = s3lib.resource_op(
+                {
+                    "obj": rgw_conn,
+                    "resource": "ObjectAcl",
+                    "args": ["nonexistingbucket", "nonexistingobject"],
+                }
+            )
+            acls_set_status = s3_obj_acl.put(ACL="private")
+
+        except ClientError as e:
+            log.info(e.response)
+            response = HttpResponseParser(e.response)
+            if (
+                response.error["Code"] == "NoSuchBucket"
+                and e.response["Error"]["Code"] == "404"
+            ):
+                log.info("Setting acl for non existing bucket failed as expected")
```

Comment:
Looks like we are net getting correct value for `e.response["Error"]["Code"] == "404"`
@vivekanandan-k-rh could you please look into this once

Reply:
@udaysk23 , With the following HTTP response, it is confirmed that it s returning 404:
2022-07-27 10:46:48,515 INFO: http reponse:
{'Error': {'Code': 'NoSuchBucket', 'BucketName': 'nonexistingbucket'}, 'ResponseMetadata': {'RequestId': 'tx00000e5e6762d631aedf5-0062e14fd8-ac64-default', 'HostId': '', 'HTTPStatusCode': 404, 'HTTPHeaders': {'content-length': '227', 'x-amz-request-id': 'tx00000e5e6762d631aedf5-0062e14fd8-ac64-default', 'accept-ranges': 'bytes', 'content-type': 'application/xml', 'date': 'Wed, 27 Jul 2022 14:46:48 GMT', 'connection': 'Keep-Alive'}, 'RetryAttempts': 0}}

And also, if the response is not 404, test case will be failed

Reply:
@vivekanandan-k-rh Thanks for changing condition parameters